File: Semestres/9.Noviembre 2020-Abril 2021/MDII/14Dic2020/Ejemplo4.go
package Ejemplo4

import "fmt"
import "math"

func recursivo(n int) float64{

  if n == 0{
    return 4.0
  }else{
    if n == 1{
      return 8.0
    }else{
      return 7.0*recursivo(n-1)+3.0*recursivo(n-2)
    }
  }
}

func formula(n int) float64{
  var A float64 = -6.0/math.Sqrt(61.0)+2.0
  var B float64 = 6.0/math.Sqrt(61.0)+2.0
  var r1 float64 =math.Pow((7.0+math.Sqrt(61.0))/2.0,float64(n))
  var r2 float64 =math.Pow( (7.0-math.Sqrt(61.0))/2.0,float64(n))

  return A*r1+B*r2

}

func main() {
  for i := 0; i<20; i++{
    fmt.Println("n ",i," recursivo ",recursivo(i)," formula ",formula(i))
  }
  
}

File: Semestres/9.Noviembre 2020-Abril 2021/MDII/22Enero2021/ejemplo4.py
import math

def recursivo(n):
  if n == 1:
    return 6
  else:
    return 3*recursivo(2*n/3)+22

def formula(n):
  return 17*pow(n,math.log(3,3/2))-11

iterator = []
for i in range(0,20):
  iterator.append(pow(3/2,i))


for i in iterator:
  print(" i ",i," Recursiva ",recursivo(i)," formula ",formula(i))
